aide-de-camp/src/runner/job_runner.rs

```diff
@@ -78,3 +163,293 @@ fn get_random_jitter() -> chrono::Duration {
         .map(|ms| chrono::Duration::milliseconds(*ms))
         .unwrap_or_else(|| chrono::Duration::milliseconds(5)) // Always takes a happy path technically
 }
+
+#[cfg(test)]
```

Comment:
The diff on this file looks extra large, but everything below here is unit tests.